index.js

```diff
@@ -1,17 +1,27 @@
 'use strict';
 const ghGot = require('gh-got');
 
-module.exports = async (username, token) => {
+const defaultOptions = {
+	headers: {
+		'user-agent': 'https://github.com/sindresorhus/gh-user'
+	}
+};
+
+module.exports = async (username, options) => {
 	if (typeof username !== 'string' || !username) {
 		throw new Error('`username` required');
 	}
 
-	const result = await ghGot(`users/${username}`, {
-		token,
-		headers: {
-			'user-agent': 'https://github.com/sindresorhus/gh-user'
+	if (typeof options === 'object') {
+		options = {...defaultOptions, ...options};
+	} else {
+		if (options !== undefined) {
+			throw new TypeError('`options` should be an object of gh-got options');
 		}
-	});
+		options = {...defaultOptions};
```

Comment:
No point in using spreading here. You can just do a normal assignment.

Reply:
I changed the signature to use a default {} options. It simplified everything.